In [1]:
import pandas as pd
from pandas import DataFrame,Series
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
import math
import random

In [2]:
from pylab import mpl
mpl.rcParams['font.sans-serif'] = ['Microsoft YaHei'] # 指定默认字体
mpl.rcParams['axes.unicode_minus'] = False # 解决保存图像是负号'-'显示为方块的问题

In [3]:
data1 = pd.read_excel(r"C:\Users\柠檬\Desktop\实验4\第6章 假设检验 .xls",header = 1,usecols = range(2))
data2 = pd.read_excel(r"C:\Users\柠檬\Desktop\实验4\第6章 假设检验 .xls",header = 1,usecols = range(2,4))

In [4]:
data2.columns=['编号','平均月支出']
frames = [data1, data2]
df1 = pd.concat(frames)
df1

编号  平均月支出
0    1    800
1    2  1 200
2    3    800
3    4  1 000
4    5    700
5    6    500
6    7  1 200
7    8  1 000
8    9  1 500
9   10    800
10  11  1 000
11  12  1 200
12  13    800
13  14    800
14  15    700
15  16    500
16  17    900
17  18  1 000
18  19  1 200
19  20    800
0   21  1 000
1   22    900
2   23  1 200
3   24    600
4   25    500
5   26    700
6   27    600
7   28  1 000
8   29  1 200
9   30  1 200
10  31    900
11  32    900
12  33  1 200
13  34    800
14  35    500
15  36    700
16  37    700
17  38  1 000
18  39  1 200
19  40  1 500

In [5]:
data1 = pd.read_excel(r"C:\Users\柠檬\Desktop\实验4\第6章 假设检验 .xls",header = 1,usecols = range(4,6))
data2 = pd.read_excel(r"C:\Users\柠檬\Desktop\实验4\第6章 假设检验 .xls",header = 1,usecols = range(6,8))

In [6]:
data1.columns=['编号','平均月支出']
data2.columns=['编号','平均月支出']
frames = [data1, data2]
df2 = pd.concat(frames)
df2

编号  平均月支出
0    1    700
1    2  1 000
2    3    800
3    4  1 200
4    5  1 200
5    6    800
6    7    600
7    8    800
8    9    900
9   10   1000
10  11    700
11  12    900
12  13    800
13  14  1 000
14  15  1 200
15  16    900
16  17    600
17  18    800
18  19    900
19  20  1 000
0   21    700
1   22  1 400
2   23    800
3   24  1 000
4   25    900
5   26  1 200
6   27  2 000
7   28  1 300
8   29    800
9   30    900
10  31    600
11  32  1 400
12  33    900
13  34  1 000
14  35    900
15  36  1 500
16  37  2 000
17  38  1 400
18  39    800
19  40    900

# T检验

## 做出假设

假设：男性大学生消费支出与女性大学生消费支出水平相近

## 处理脏数据（部分数据带有空格）

In [7]:
import types
men = []
for x in df1['平均月支出'].values:
    if(type(x) is not int):
        x = ''.join(x.split())
    men.append(x)
# df1['平均月支出'].values = ''.join(df1['平均月支出'].values.split())
men = list(map(int, men))
men

[800,
 1200,
 800,
 1000,
 700,
 500,
 1200,
 1000,
 1500,
 800,
 1000,
 1200,
 800,
 800,
 700,
 500,
 900,
 1000,
 1200,
 800,
 1000,
 900,
 1200,
 600,
 500,
 700,
 600,
 1000,
 1200,
 1200,
 900,
 900,
 1200,
 800,
 500,
 700,
 700,
 1000,
 1200,
 1500]

In [8]:
women = []
for x in df2['平均月支出'].values:
    if(type(x) is not int):
        x = ''.join(x.split())
    women.append(x)
women = list(map(int, women))
women

[700,
 1000,
 800,
 1200,
 1200,
 800,
 600,
 800,
 900,
 1000,
 700,
 900,
 800,
 1000,
 1200,
 900,
 600,
 800,
 900,
 1000,
 700,
 1400,
 800,
 1000,
 900,
 1200,
 2000,
 1300,
 800,
 900,
 600,
 1400,
 900,
 1000,
 900,
 1500,
 2000,
 1400,
 800,
 900]

## 进行levene检验，检验两总体是否具有方差齐性

In [9]:
stats.levene(men, women)

LeveneResult(statistic=0.024874211607965418, pvalue=0.8750884814724933)

p值远大于0.05，认为两总体具有方差齐性。

## 进行两独立样本T检验

In [10]:
stats.ttest_ind(men,women)

Ttest_indResult(statistic=-1.3236759213525044, pvalue=0.18947719784107386)

p值为0.18947719784107386，大于α（0.05），所以接受原假设
结论为：男性大学生消费支出与女性大学生消费支出水平相近

# Z检验

## 作出假设

与上述假设相同

## 进行两独立样本Z检验

In [11]:
import statsmodels.stats.weightstats as sw
sw.ztest(men, women, value=0)

(-1.3236759213525044, 0.18561069860827706)

p值为0.18561069860827706，大于α（0.05），所以接受原假设
结论为：男性大学生消费支出与女性大学生消费支出水平相近